### PR TITLE
Move test environment out of the way

### PR DESCRIPTION
**Pull request by portante**
Isolate the test infrastructure from the rest of the server code.

The benefit will be towards merging this the remains of the `b0.69` branch into a fully deployable `b0.72` branch where the legacy server code and the new server code can run side-by-side without the need of 2 separate branches to coordinate code changes.

To facilitate an easier review, the file moves/renames are in one commit, and the 3 files that have to change to use the files in their new locations are in the second commit.

See https://github.com/portante/pbench/pull/23 for the proposal for creating a `b0.72` branch using this migration.

### CODE REVIEW

**Review by portante — COMMENTED** [review]
Ready for review.

File: server/test/run [hunk 1]
```diff
@@ -13,9 +13,19 @@ unset _PBENCH_SERVER_CONFIG
...
+export _ldir=${_server}/lib
+test -d "${_ldir}" || exit 1
+export _tdir=${_server}/test
+test -d "${_tdir}" || exit 1
+_webs=$(realpath -e ${_server}/../web-server)
+if [[ ! -d "${_webs}" ]]; then
+    echo "ERROR: failed to resolve legacy web-server directory, \"${_webs}\"" >&2
+    exit 1
+fi
+export _webs
```
Comment (webbnh) [inline server/test/run:28]: Why do you defer `export`'ing `_webs` when you didn't defer exporting any of the preceding variables?

And, if this script exits suddenly with a `1`, are we going to wish that the `_?dir` variables had had nice error messages like `_webs` does?

**Review by webbnh — APPROVED** [review]
Looks OK to me, although there's a nit....

File: server/test/run (hunk 1 above)
Comment (portante) [reply server/test/run]: > Why do you defer `export`'ing `_webs` when you didn't defer exporting any of the preceding variables?

No good reason, just missed it when I changed from a direct test to using `realpath`.  Fixed.

> And, if this script exits suddenly with a `1`, are we going to wish that the `_?dir` variables had had nice error messages like `_webs` does?

Yes, I had that change in another PR, so I've added here instead.

File: server/test/run [hunk 2]
```diff
@@ -13,26 +13,24 @@ unset _PBENCH_SERVER_CONFIG
...
-export _server=$(dirname $(dirname $(realpath ${0})))
+function _log_exit {
+    echo "${1}" >&2
+    exit 1
+}
+
+export _server=$(dirname $(dirname $(realpath "${0}")))
 export _bdir=${_server}/bin
-test -d "${_bdir}" || exit 1
+test -d "${_bdir}" || _log_exit "ERROR: Missing \"${_bdir}\""
```
Comment (dbutenhof) [inline server/test/run:23]: OK, so this comes out something like `ERROR: Missing "server/bin"`, but we're doing a `-d` check not a `-e` check; so maybe something more like `ERROR: ${_bdir} missing or not a directory`??

**Review by dbutenhof — DISMISSED** [review]

File: server/test/run (hunk 2 above)
Comment (portante) [reply server/test/run]: Fixed.

**Review by dbutenhof — APPROVED** [review]

**Review by ndokos — COMMENTED** [review]
LGTM

**Pull request merged**